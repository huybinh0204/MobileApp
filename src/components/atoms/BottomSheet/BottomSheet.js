import PropTypes from 'prop-types';
import React, { useEffect, useRef } from 'react';
import RNBottomSheet from 'react-native-raw-bottom-sheet';

const BottomSheet = ({ isVisible, children, onClose, styles, height, onOpen, ...rest }) => {
  const sheetRef = useRef(null);

  useEffect(() => {
    isVisible ? sheetRef.current?.open() : sheetRef.current?.close();
  }, [isVisible]);

  return (
    <RNBottomSheet
      ref={sheetRef}
      height={height}
      onOpen={onOpen}
      onClose={onClose}
      customStyles={styles}
      {...rest}
    >
      {children}
    </RNBottomSheet>
  );
};

BottomSheet.defaultProps = {
  isVisible: false,
  styles: {},
  height: 400,
};

BottomSheet.propTypes = {
  isVisible: PropTypes.bool,
  children: PropTypes.element.isRequired,
  height: PropTypes.number,
  onClose: PropTypes.func,
  onOpen: PropTypes.func,
  styles: PropTypes.shape({
    wrapper: PropTypes.object,
    container: PropTypes.object,
    draggableIcon: PropTypes.object,
  }),
};

export default BottomSheet;
