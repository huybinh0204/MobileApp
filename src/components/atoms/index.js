export { default as AccountNumber } from './AccountNumber/AccountNumber';
export { default as Badge } from './Badge/Badge';
export { default as Balance } from './Balance/Balance';
export { default as BaseFormInput } from './BaseFormInput/BaseFormInput';
export { default as BaseText } from './BaseText/BaseText';
export { default as BottomSheet } from './BottomSheet/BottomSheet';
export { default as Card } from './Card/Card';
export { default as CurrencyInput } from './CurrencyInput/CurrencyInput';
export { default as Dash } from './Dash/Dash';
export { default as Form } from './Form/Form';
export { default as FormInput } from './FormInput/FormInput';
export { default as FormSubmitButton } from './FormSubmitButton/FormSubmitButton';
export { default as GeneralContainer } from './GeneralContainer/GeneralContainer';
export { default as IconSvg } from './IconSvg/IconSvg';
export { default as Link } from './Link/Link';
export { default as MainButton } from './MainButton/MainButton';
export { default as Money } from './Money/Money';
export { default as PdfView } from './PdfView/PdfView';
export { default as Plaid } from './Plaid/Plaid';
export { default as PressableListItem } from './PressableListItem/PressableListItem';
export { default as ProgressBar } from './ProgressBar/ProgressBar';
export { default as SegmentedTextInput } from './SegmentedTextInput/SegmentedTextInput';
export { default as SelectButton } from './SelectButton/SelectButton';
export { default as Switch } from './Switch/Switch';
export { default as TabItem } from './TabItem/TabItem';
export { default as Toast } from './Toast/Toast';
export { default as TransactionItem } from './TransactionItem/TransactionItem';
