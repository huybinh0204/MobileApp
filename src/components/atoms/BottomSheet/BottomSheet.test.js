import { render } from '@testing-library/react-native';
import React from 'react';
import { BaseText } from '_components/atoms';
import BottomSheet from './BottomSheet';

describe('BottomSheet', () => {
  let props;

  beforeEach(() => {
    props = {
      isVisible: true,
      children: <BaseText>Hi</BaseText>,
      onClose: jest.fn(),
      onOpen: jest.fn(),
    };
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<BottomSheet {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call onOpen if isVisible is true', () => {
    render(<BottomSheet {...props} />);
    expect(props.onOpen).toBeCalled();
    expect(props.onClose).not.toBeCalled();
  });

  it('should render the content', () => {
    const { getByText } = render(<BottomSheet {...props} />);
    expect(getByText('Hi')).not.toBeNull();
  });
});
