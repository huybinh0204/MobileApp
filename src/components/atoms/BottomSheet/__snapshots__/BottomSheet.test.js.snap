// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheet should match the snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "portrait-upside-down",
      "landscape",
      "landscape-left",
      "landscape-right",
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#00000077",
            "flex": 1,
          },
          undefined,
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      collapsable={false}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 0,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <Text
        allowFontScaling={false}
      >
        Hi
      </Text>
    </View>
  </View>
</Modal>
`;
